sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -1,48 +1,112 @@
-    seq = sbol3.Sequence(identity=record.accession[0] + "_sequence",
-                    elements=record.sequence.lower(),
-                    encoding=sbol3.IUPAC_DNA_ENCODING)
-    doc.add(seq)
-    comp.sequences = [seq]
-    if write: 
-        doc.write(fpath=sbol3_file, file_format=sbol3.SORTED_NTRIPLES)
-    return doc
+TEST_NAMESPACE = "https://test.sbol3.genbank/"
+# TODO: Temporarily assuming only dna components to be dealt with in genbank files
+COMP_TYPES = [sbol3.SBO_DNA]
+# TODO: Temporarily assuming components to only have the engineered_region role
+COMP_ROLES = [sbol3.SO_ENGINEERED_REGION]
+# TODO: Need to find a way to infer SequenceFeature Range/Cut Orientation from the genbank file
+SEQ_FEAT_RANGE_ORIENTATION = "https://identifiers.org/SO:0001030"  # inline orientation
+# TODO: Temporarily encoding sequnce objects in IUPUC mode only
+SEQUENCE_ENCODING = sbol3.IUPAC_DNA_ENCODING
+# Complete file paths to pass as paramteres for testing
+SAMPLE_GENBANK_FILE_1 = os.path.join(os.path.abspath(os.path.curdir), 
+                                "test/test_files", "BBa_J23101.gb")
+SAMPLE_SBOL3_FILE_1 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "BBa_J23101_from_genbank_to_sbol3_direct.nt")
+SAMPLE_GENBANK_FILE_2 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "iGEM_SBOL2_imports.gb")
+SAMPLE_SBOL3_FILE_2 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "iGEM_SBOL2_imports_from_genbank_to_sbol3_direct.nt")
+GB2SO_MAPPINGS = {}
+SO2GB_MAPPINGS = {}
+SO_SEQ_FEAT_ROLE_NS = "http://identifiers.org/so/"
+GB2SO_MAPPINGS_CSV = os.path.join(os.path.abspath(os.path.curdir), "gb2so.csv")
+SO2GB_MAPPINGS_CSV = os.path.join(os.path.abspath(os.path.curdir), "so2gb.csv")
+
+
+class GenBank_SBOL3_Converter:
+    def create_GB2SO_role_mappings(self, gb2so_csv: str = GB2SO_MAPPINGS_CSV, so2gb_csv: str = SO2GB_MAPPINGS_CSV):
+        """Reads 2 CSV Files containing mappings for converting between GenBank and SO ontologies roles
+        :param gb2so_csv: path to read genbank to so conversion csv file
+        :param so2gb_csv: path to read so to genbank conversion csv file
+        """
+        with open(gb2so_csv, mode='r') as csv_file:
+            csv_reader = csv.DictReader(csv_file)
+            for row in csv_reader:
+                GB2SO_MAPPINGS[row["GenBank_Ontology"]] = row["SO_Ontology"]
+        with open(so2gb_csv, mode='r') as csv_file:
+            csv_reader = csv.DictReader(csv_file)
+            for row in csv_reader:
+                SO2GB_MAPPINGS[row["SO_Ontology"]] = row["GenBank_Ontology"]
+
+
+    def convert_genbank_to_sbol3(self, gb_file: str, sbol3_file: str = "sbol3.out", namespace: str =
+                                TEST_NAMESPACE, write: bool = False) -> sbol3.Document:
+        """Convert a GenBank document on disk into an SBOL3 document
```

Comment:
Great job with the docstring here!

Reply:
Thanks! I realized there would need to be many more helper methods like create_mappings, so making them into a class with docstrs would be clean.